drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -160,6 +151,13 @@ static const struct stm32_temp_config stm32_temp_dev_config = {
 static struct stm32_temp_data stm32_temp_dev_data = {
 	.adc = DEVICE_DT_GET(DT_INST_IO_CHANNELS_CTLR(0)),
 	.channel = DT_INST_IO_CHANNELS_INPUT(0),
```

Comment:
can be removed, seems duplicate to .channel_id in adc_cfg

Reply:
This `channel` variable in the `struct stm32_temp_data` indeed is not very useful, but this line is added by you. I can probably remove them altogether

Reply:
I have removed the `channel` variable from the struct